tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -1857,7 +1857,8 @@ IrEmitterUnnested::BuildKernelThunkFromBufferSlices(
     absl::string_view name, Thunk::ThunkInfo thunk_info,
     absl::Span<const BufferSlice* const> slices,
     std::function<void(const BufferSlice*, llvm::Value*)>
-        bind_slice_to_ir_value) {
```

Comment:
Boolean params to functions which change behavior are usually an antipattern, is there a way to avoid this?

Reply:
How about let's simply set the first seen temp buffer to `temp_buffer` (and remove the `LOG(FATAL)` below)? Or any other suggestions?

I really don't see any codes use this set temp buffer in the public repo, so I don't know what the best way to revise it is.

The only related code I see is that the kernel function arguments require the last argument to be a temp buffer pointer.

Reply:
I don't see a good way to remove this Boolean param.

One thing we can do is to remove the codes searching for temp buffer and always set `temp_buffer` to nullptr, as we don't expect any users of this pointer.

Please let me know what you think.

Reply:
But `temp_buffer` is inserted into `buffers_needed`? I think I'll be fine with just removing both the param and `LOG(FATAL)`.

Actually I'm not sure I fully understand the logic of what's going on when we have multiple `temp_buffers`: would this loop just insert the last one?

Reply:
Modified according to the suggestion. It latches the first seen temp buffer (comments added in the code to make it explicit).

Yes, the `temp_buffer` is inserted into `buffers_needed`, so it might affect the kernel function signature. However, my read is still that the main part is `bindings_.SetTempBufferBase(kernel_args.at(*temp_buffer));` and so long as no one invokes `GetTempBufferBase()` subsequently, the set `temp_buffer` pointer is never used.

If what I read is correct, it should be better to remove the related codes. That is removal of `SetTempBufferBase()` and `GetTempBufferBase()` and change of the kernel function signature.

Reply:
Though removing it in this one should work as well.

Reply:
No problem, I will do a separate PR to clean it. I was quite bothered by this `temp_buffer`.